roadmap/propagate-rust-panic-through-native-frame.md

```diff
@@ -46,3 +48,37 @@ cases (contains destructors, no affordances). We will also have to
 specify this on a per-target basis, as the details will vary depending
 on what exception mechanism is in use.
 
+Another consideration is whether the the native frames include a `catch` block.
+There are several possibilities here as well:
+
+* "catch native" -- e.g. a C++ block catching a specific subset of C++ exceptions
+* "catch all" -- e.g. a C++ catch without an exception argument
+* "catch and release" -- e.g. a C++ "catch all" that re-`throw`s the exception
+
+So far we do **not** have plans to specify these cases.
+
+## Unwind-termination and related concerns
+
+There are several possibilities for the top section of the diagram, above the
+native frames:
+
+* Rust frames - the `panic` re-enters Rust frames
+  * At this point, the `panic` will be no different from a normal `panic`,
+    unless and until it re-enters native frames.
```

Comment:
I'm not sure what you mean - are you saying that the thread- and app- boundary cases don't need to be stated?

Reply:
More like all these guarantees can be stated by just saying that: "When an `extern "C unwind" { fn foo(); }` declaration unwinds and the unwinding was started with a Rust panic, that unwinding proceeds exactly as if it would have happened in Rust". 

This means that all other bullet points then just follow from how `panic!` is currently guaranteed to work in Rust.